helper/include/odp/helper/threads.h

```diff
@@ -261,16 +275,40 @@ int odph_thread_create(odph_thread_t thread[],
  * A function call may be used to wait any number of launched threads to exit.
  * A particular thread may be waited only once.
  *
+ * Threads are joined in the order they are in 'thread' table. Returns on the
+ * first non-zero exit status or other failure.
+ *
  * @param thread        Table of threads to exit
  * @param num           Number of threads to exit
  *
- * @return Number of threads exited
+ * @return Number of threads successfully joined with zero exit status
+ *         (0 ... num)
  * @retval -1  On failure
  *
  * @see odph_thread_create()
  */
 int odph_thread_join(odph_thread_t thread[], int num);
 
+/**
+ * Wait previously launched threads to exit
+ *
+ * Similar to odph_thread_join() but outputs results of joined threads and
+ * stops only if the actual join operation fails for some thread. Threads are
+ * joined in the order they are in 'thread' table. Returns number of threads
+ * successfully joined and writes respective exit statuses into the 'res'
+ * table.
+ *
+ * @param thread        Table of threads to exit
+ * @param[out] res      Table for result output
+ * @param num           Number of threads to exit and results to output
+ *
+ * @return Number of threads successfully joined (0 ... num)
+ * @retval -1  On failure
```

Comment:
Could be "invalid parameters", since that's the only case in which -1 is returned.